src/hooks/useBalances.ts

```diff
@@ -109,30 +110,49 @@ async function _getBalances(walletInfo: WalletInfo, tokens: TokenDetails[]): Pro
   return balances.filter(Boolean) as TokenBalanceDetails[]
 }
 
-export const useTokenBalances = (): UseTokenBalanceResult => {
+export const useBalances = (): UseBalanceResult => {
   const walletInfo = useWalletConnection()
+  const [ethBalance, setEthBalance] = useSafeState<BN | null>(null)
   const [balances, setBalances] = useSafeState<TokenBalanceDetails[]>([])
   const [error, setError] = useSafeState(false)
 
   const tokens = useTokenList(walletInfo.networkId)
 
+  // Get token balances
   useEffect(() => {
-    // can return NULL (if no address or network)
-    walletInfo.isConnected &&
+    if (walletInfo.isConnected) {
       _getBalances(walletInfo, tokens)
         .then(balances => {
           logDebug(
-            '[useTokenBalances] Wallet balances',
+            '[useBalances] Wallet balances',
             balances ? balances.map(b => formatAmount(b.walletBalance, b.decimals)) : null,
           )
           setBalances(balances)
           setError(false)
         })
         .catch(error => {
-          console.error('[useTokenBalances] Error loading balances', error)
+          console.error('[useBalances] Error loading token balances', error)
           setError(true)
         })
+    }
   }, [setBalances, setError, walletInfo, tokens])
 
-  return { balances, error, tokens }
+  // Get ether balances
+  useEffect(() => {
+    if (walletInfo.isConnected) {
+      walletApi
+        .getBalance()
```

Comment:
Do we really need this call everywhere we use tokenBalances? Wouldn't it be better in its own hook that will be used only in /wallet, only for WETH, otherwise wasted extra request

Reply:
Probably, I'll try to divide the hook

Reply:
Addressed here dfba6c26fb761b0ffecdeb4b85e276a040f7c3b1